flink-ml-servable-core/src/main/java/org/apache/flink/ml/linalg/Vector.java

```diff
@@ -24,29 +24,29 @@
 
 import java.io.Serializable;
 
-/** A vector of double values. */
+/** A vector representation of numbers. */
 @TypeInfo(VectorTypeInfoFactory.class)
 @PublicEvolving
-public interface Vector extends Serializable {
+public interface Vector<K extends Number, V extends Number> extends Serializable {
 
     /** Gets the size of the vector. */
-    int size();
+    K size();
 
     /** Gets the value of the ith element. */
-    double get(int i);
+    V get(K i);
 
     /** Sets the value of the ith element. */
-    void set(int i, double value);
+    void set(K i, V value);
 
-    /** Converts the instance to a double array. */
-    double[] toArray();
+    /** Converts the instance to a primitive array. */
+    Object toArray();
 
     /** Converts the instance to a dense vector. */
-    DenseVector toDense();
+    Vector<K, V> toDense();
 
     /** Converts the instance to a sparse vector. */
-    SparseVector toSparse();
+    Vector<K, V> toSparse();
```

Comment:
The intention of defining this method in an interface is to have the caller use the return value via public APIs of `Vector<K, V>`.

However, it appears that caller of this method uses return values via public fields/APIs that do not belong to `Vector<K, V>`. For example, `MinHashLSHModelData#hashFunction()` accesses the field `SparseIntDoubleVector#indices` via `vec.toSparse().indices`. The code will break if we change the concrete class from `SparseIntDoubleVector` to `SparseLongDoubleVector`.

Reply:
I have overrode the return type in `IntDoubleVector` and `LongDoubleVector`, since for different instances `toSparse` should return different types of vectors.

We can talk about this offline.